mlrun/projects/project.py

```diff
@@ -2424,9 +2424,18 @@ def _instantiate_function(
             )
 
         if isinstance(func, str):
-            # in hub or db functions name defaults to the function name
-            if not name and not (func.startswith("db://") or func.startswith("hub://")):
-                raise ValueError("Function name must be specified")
+            # In hub or db functions name defaults to the function name
+            # If name is specified, we try to create a new function with this name.
```

Comment:
I suggest moving this comment `# If name is specified, we try to create a new function with this name.` to before line 3439, where the function is actually created.